### ISSUE TITLE
Factual information in successful petitions is ignored

### ISSUE THREAD

**Issue by coryalder** [WhiteHouse/petitions#2]
Steps to repeat:
1. Create a petition pointing out a number verifiable facts related to something wrong with government or how America is run.
2. Get many signatures, enough to cross the current petition threshold.
3. Receive a letter from the White House blatantly ignoring or contradicting the factual information in the petition, and explaining the pre-existing position of the administration.
4. Have approximately the same petition be repeated periodically, because people are still upset about the issue because the White House hasn't actually _done_ anything about the issue.
### For example

https://petitions.whitehouse.gov/petition/abolish-tsa-and-use-its-monstrous-budget-fund-more-sophisticated-less-
intrusive-counter-terrorism/c7L94bFB?utm_source=wethepeople&utm_medium=response&utm_campaign=tsa

**Petition body:**

"It has failed approximately 70% of undercover efficacy tests, and for all its excesses, has been unable to catch even a single terrorist since its creation."

**Petition response:**

"Over the past 10 years, TSA has strengthened security by creating successful programs and deploying technologies that were not in place prior to September 11, while also taking steps whenever possible to enhance the passenger experience."

**Comment by ceejayoz** [comment]
This is hardly an issue with the codebase.

**Comment by brycec** [comment]
It is, however, a design issue with the project as a whole. How can this be a useful tool otherwise?

**Comment by christiangenco** [comment]
Here's a fixed link to the example petition above to [Abolish the TSA, and use its monstrous budget to fund more sophisticated, less intrusive counter-terrorism intelligence](https://petitions.whitehouse.gov/petition/abolish-tsa-and-use-its-monstrous-budget-fund-more-sophisticated-less-
intrusive-counter-terrorism/c7L94bFB?utm_source=wethepeople&utm_medium=response&utm_campaign=tsa), as well as an infamous petition to [Legalize and Regulate Marijuana in a Manner Similar to Alcohol](https://petitions.whitehouse.gov/petition/legalize-and-regulate-marijuana-manner-similar-alcohol/y8l45gb1).

**Comment by stephenroller** [comment]
Github is probably not the place for such a debate. The authors of this software are not the President.

**Comment by eykanal** [comment]
So developers working for the white house perform the admirable step of open sourcing some of their code, and the first thing you do is attempt to derail their efforts by making a stupid politically motivated comment? Please don't; you make the entire developer community look bad by doing this.

**Comment by harshavardhana** [comment]
Agreed with @eykanal - github is not a medium nor a tool for making these debates

**Comment by gluster-eco** [comment]
I agree with @Harshavardhana and @eykanal as well, this is a huge step forward for our government and our ability to act as a real democracy, in something like real time.  If you want to provide farcical and inane commentary, at least make it fun a la the following example:

https://bugzilla.gnome.org/show_bug.cgi?id=626593

**Comment by brycec** [comment]
What if we discuss ways in which we can improve the design of this platform so that users have a mechanism of dealing with failed petitions?

For example, implementing a way to re-open a petition with a counter-response to the government, allowing for additional signatures.

I don't think it's simply a political issue that petitions can basically just be ignored on this platform.

**Comment by imdabes** [comment]
> Goals
> 
> President Obama is committed to creating the most open and participatory government in our nation’s history, and this petitioning platform is a key part of that initiative.

This isn't just a release of the underlying software, it isn't separated from the political campaign in any capacity due to the obvious tie-ins within the main readme / goal statement. 

If the goal was simply to create a petition system, it would be a fundamentally different proposal and such political criticism would be off-topic, however this is not the case.

**Comment by mattstreet** [comment]
As long as we don't spam this repo, I think one issue makes a good point. But I agree that since it's not going to help change things at all, lets try to limit discussion to this one issue and not make any more.

**Comment by eykanal** [comment]
@brycec - That sounds like a much better approach to me, but I don't think this particular issue can be salvaged. I'd close this one and open an new one.

Do realize that your proposed issue would likely require a large amount of development effort.

**Comment by anildash** [comment]
This is a great starting point for turning this discussion from trolling
into something productive.

One related technique that may work, if people feel there are incorrect
statements being made in responses, would be to add the ability to annotate
an official response with documented links, which could then be displayed
if they got enough votes.

On Thursday, August 23, 2012, Bryce Culhane wrote:

> What if we discuss ways in which we can improve the design of this
> platform so that users have a mechanism of dealing with failed petitions?
> 
> For example, implementing a way to re-open a petition with a
> counter-response to the government, allowing for additional signitures.
> 
> I don't think it's simply a political issue that petitions can basically
> just be ignored on this platform.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/WhiteHouse/petition/issues/2#issuecomment-7989252.

## 

---

about: http://www.anildash.com/
im/twitter: anildash phone: +1 646 833-8659

**Comment by mattstreet** [comment]
Why not fork this project and add some basic functionality that makes it easier to pretend what people think? The challenge - response system sounds like a good idea.

**Comment by mattstreet** [comment]
@anildash - I like that idea, maybe something peer moderated? Sorta like how wikipedia works?

**Comment by GovInTrenches** [comment]
Civic hackers serve the community and not any particular elected official. I think that should probably be kept in mind when making github issues. I don't think trying to use github issues to make a political point is productive. 

I do agree with @anildash and @brycec's though.

**Comment by mattstreet** [comment]
That's why I think limiting this to one issue that just stays open is a good idea. The developers can either let it sit unfixed and eventually the long standing unfixed issue really stands out, but they can otherwise ignore  it, or they can junk it or declare it fixed and people will get made and keep reissuing it.

Or (fat chance) it could be fixed, or at least discussed honestly. Wouldn't it be amazing if politician's just admitted that our system requires them to be full of it?

**Comment by imdabes** [comment]
@GovInTrenches

This repo was politicized by the software authors in the first place though

**Comment by mattstreet** [comment]
Good point @GovInTrenches.

**Comment by GovInTrenches** [comment]
@imdabes Perhaps, but the software's been opensource'd now. It belongs to the community and as the community we can bend it towards a 'the-code-has-no-party' philosophy. 

It's not just We The People that would be good to see open sourced. Personally, I'd like to see this app: http://thehill.com/blogs/hillicon-valley/technology/217063-cantor-launches-citizen-cosponsor-facebook-app open sourced as well. 

I'd hate to live in a world where civic apps are divided by party lines

**Comment by imdabes** [comment]
In that case can we change readme.md to remove non-factual information?

**Comment by max-mapper** [comment]
This issue should be closed and specific issues should be opened that are constructive in specific ways

**Comment by max-mapper** [comment]
also, trolling OP is canadian

**Comment by mattstreet** [comment]
@maxogden And? The US government concerns itself with every other government out there if it benefits or threatens them. Why shouldn't our neighbors to the north be concerned about our democracy?

**Comment by max-mapper** [comment]
@mattstreet http://en.wikipedia.org/wiki/Troll_(Internet)

**Comment by williscool** [comment]
I think this issue will require some upstream changes

**Comment by unirgy** [comment]
This is a valid BLOCK issue. What's the point wasting time in helping to develop/test the software if its only purpose is a red herring and political theater?

**Comment by amoffat** [comment]
I second @unirgy's point

I don't think anyone here is unappreciative of the effort put forth by the developers to create and open source this.  Many of us know that it takes guts to put your work up for criticism and engagement.  But I think it's important that all contributors should know what this software is being used for before they contribute.

A petition, according to dictionary.com is:

> a formally drawn request, often bearing the names of a number of those making the request, that is addressed to a person or group of persons in authority or power, soliciting some favor, right, mercy, or other benefit

The implication here is that the authority is being stimulated into some form of action by the will of many people.  But the We The People project has made it clear that these petitions do not result in action...just "official responses" as to why things are they way they are.  It's political theater.  Faux democracy.  Even the rhetoric of the project, "We The People", has a false sense of empowerment, as if because "we the people" craft a petition, it has the power to change things.

This is all very important because developers may spend their valuable time contributing to a project that they think is helping open source democracy, when in reality they're contributing to the political theater.  So yes, the issue some of us are taking isn't with the software _per se_, but it is with how the software will be immediately used, and so therefore I cast my vote that this issue is important.

**Comment by richo** [comment]
@GovInTrenches Since when does open sourcing code make it "the communities' property" ?

I open source tons of code, and the community owns none of it. You're welcome to inspect, modify and use it as you see fit, but that doesn't confer ownership.

As an aside, there are bugger all projects for which democracy works, and the ones that do are huge (think apache, python)

**Comment by disassembler** [comment]
While I agree with the original poster in everything he says, this is not the correct medium for such a debate. If you're going to report an issue, find one relevant to the forum it is being reported in. That being said though, I was very impressed with the original posters detail in his steps to reproduce the bug ;-)

**Comment by unirgy** [comment]
@tombell @disassembler thank you for your opinion. 

What makes this issue effectively valid is the fact that many people believe that this issue is valid and belongs here. 

If, after reading the issue and comments here, you still think that this issue should not be brought to the attention of potential contributors in the most visible manner, you can just merrily ignore it, as no one here really forces you to accept their opinions.

This issue is a factual and informative warning to potential contributors who will appreciate the information and take it to their attention.

**Comment by brycec** [comment]
@tombell You're missing the point. It's not an issue with the way the project is being used. It's an issue with the _design_ of the project. If the goal of the project is to give people a voice, then users should not feel helplessly ignored. In software terms, it is _bad UX_. There are actionable things we can do to the project to help with this. Please see my previous comment and also @anildash

**Comment by paul-m** [comment]
Organizations which adopt the use of this software tool would need to have processes in place to make sure the results aren't ignored.

That issue can't be addressed in a pull request for the software itself.

**Comment by amoffat** [comment]
> I would love to see how you are going to make an application control human behaviour of the human that uses this software. That would be amazing.

@tombell Influencing behavior is not that hard.  Remove the ability for an organization to respond with "official responses."  Replace it such that the organization must respond with a plan to bring the issue to a wider audience, with the purposes of gathering "votes."  Give this response/plan a progress meter so the original petition signers can keep track of the organization's progress.

**Comment by amoffat** [comment]
@tombell You asked for an application that can influence behavior on the issue.  Removing official responses and replacing it with _something else_ that is more proactive will do that.

I'll try to find time to get to it today, unless someone else wants to get to it first.

**Comment by paul-m** [comment]
@amoffat How about a patch or pull request?

**Comment by amoffat** [comment]
@paul-m yeah that's what we should do.  I've never used Drupal, and it's been years since I've touched PHP.  Anyone have any tips to diving in quickly to make this change?

**Comment by Bastlynn** [comment]
Fork it - then ideally set it up your code as it's own module, so that those setting up the system can pick which approach is best suited to their situation without removing the prior behavior that existing orgs would want to keep. By D.O. convention at least, that's most likely to be the approach that gets your code pulled in and in use. Once code is available that makes supporting the business process you want to see more likely - then it's more likely you'll see the business practice adopted by newer organizations.

Depending on the spec you're looking for in this - it could even be set up as a combination of Rules and Views and Features without even touching code. (See those modules on D.O. as needed, Features is used to package configuration for release in a module.) Spec is probably your first step though. Figure out what the steps to get what you want done actually are then go from there.

**Comment by saizai** [comment]
FWIW, if you're interested in working on open source code that isn't going to just result in a brushoff, ping me. I'm working on makeyourlaws.org - which is the toothful, third party, no begging version of this.

**Comment by aawwawa** [comment]
Yeah, really.  You open up the code, presumably hoping that it will get some review and contributions from the community, but what's it going to be used for? To project an illusion of progressiveness on the administration's part?  Anyone who contributes to this code is horribly misguided and completely indoctrinated.  This is a total fucking joke.

**Comment by Bastlynn** [comment]
At this point, I wouldn't consider the White House the only client base here. It's open source now, which means it's up for use by anybody. Regardless of the origin of the original code base - it has the potential to be extremely useful to other legislative bodies or to non-governmental organizations. Which is more important to you? Getting the job done or talking about it? Do you want to see the work used for good (even if by other hands) or not?

**Comment by Mathnerd314** [comment]
There is a factual response, namely "The statistic cited in the petition was from testing performed nearly eight years ago and doesn't reflect the current security environment. Since then, TSA has implemented new security measures and deployed enhanced technology to address evolving threats to aviation."

**Comment by paul-m** [comment]
If anyone has a solution _in code_ to this problem, they should please go ahead and send a pull request. Here's how you do it: https://help.github.com/articles/using-pull-requests

**Comment by gagarine** [comment]
I think their is some way to fix a bit this problem:
- First it should not be possible to delete anythings. You can say "it's transparent" and delete stuff in an unilateral way.
- You need a different closing state (like for bug duplicate, fixed, postpone, ... but with a different vocabulary..)
- Even if somethings is "closed" peoples should still be able to comment and discuss. 
- Front page should show closed petition than still have a lot of activities.
- Gov need to close issue with better explanation, add a required "source" field.

...they said "The media is the message".

**Comment by Bastlynn** [comment]
Agreed re: deletion. Marking something as closed probably should close the conversation on it, simply for sanity's sake in moderation efforts and to avoid people trying to push on a closed project instead of opening a new one. Otherwise "closed" actually doesn't mean anything. Perhaps a 'resurrect' option if enough people vote to bring it back to the table? Either way, I agree that the record of the effort, comments on it, and the like - should remain publicly viewable as a transparent record.

I'm curious how you would qualify the sources field outside of a WH context. Some things just don't have a source to cite reference? What's the goal behind adding a source field?

**Comment by chadwhitacre** [comment]
I am able to reproduce the bug locally.

:+1: to @unirgy.

**Comment by chadwhitacre** [comment]
They [listened to us](https://petitions.whitehouse.gov/response/ale-chief-white-house-beer-recipe)! Oh wait ...
